src/core/canonical_constructor.jl

```diff
@@ -1,40 +1,81 @@
 
-function _canonical_model_init(bus_count::Int64,
+function _container_spec(V::DataType, ax...; kwargs...)
+
+    if :parameters in keys(kwargs)
+        parameters = kwargs[:parameters]
+    else
+        parameters = true
+    end
+
+    # While JuMP fixes the isassigned problems
+    if parameters
+            cont = JuMP.Containers.DenseAxisArray{PGAE{V}}(undef, ax...)
+            _remove_undef!(cont.data)
+        return cont
+    else
+            cont = JuMP.Containers.DenseAxisArray{GAE{V}}(undef, ax...)
+            _remove_undef!(cont.data)
+        return cont
+    end
+    
+
+    #=
+    if parameters
+        return JuMP.Containers.DenseAxisArray{PGAE{V}}(undef, ax...)
+    else
+        return JuMP.Containers.DenseAxisArray{GAE{V}}(undef, ax...)
+    end
+    =#
+end
+
+function _canonical_model_init(bus_numbers::Vector{Int64},
                               optimizer::Union{Nothing,JuMP.OptimizerFactory},
                               transmission::Type{S},
-                              time_periods::Int64; kwargs...) where {S <: PM.AbstractPowerFormulation}
+                              time_range::UnitRange{Int64}; kwargs...) where {S <: PM.AbstractPowerFormulation}
+
+    if :parameters in keys(kwargs)
+        parameters = kwargs[:parameters]
+    else
+        parameters = true
+    end
 
     jump_model = _pass_abstract_jump(optimizer; kwargs...)
     V = JuMP.variable_type(jump_model)
+
     ps_model = CanonicalModel(jump_model,
                             Dict{Symbol, JuMP.Containers.DenseAxisArray}(),
                             Dict{Symbol, JuMP.Containers.DenseAxisArray}(),
-                            zero(JuMP.GenericAffExpr{Float64, JuMP.variable_type(jump_model)}),
-                            Dict{Symbol, JuMPPGAEArray{V}}(:var_active => JuMPPGAEArray{V}(undef, bus_count, time_periods),
-                                                                        :var_reactive => JuMPPGAEArray{V}(undef, bus_count, time_periods)),
-                            Dict{Symbol,Any}(),
+                            zero(JuMP.GenericAffExpr{Float64, V}),
+                            Dict{Symbol, JuMP.Containers.DenseAxisArray}(:var_active => _container_spec(V, bus_numbers, time_range; kwargs...),
+                                                                         :var_reactive => _container_spec(V, bus_numbers, time_range; kwargs...)),
+                            parameters ? Dict{Symbol,JuMP.Containers.DenseAxisArray}() : nothing,
```

Comment:
It seems like you could avoid passing "parameters" throughout the stack like you're doing.  It's already implicitly part of CanonicalModel.  You just need to check the type of ps_m.parameters, which you could put in a helper function.

Also, I don't see any documentation for callers about what passing in "parameters" is going to do.

Reply:
I think is only applicable sometimes depending on when the parameters are created. Will leave it for a future PR to check where is good to use.